recipes/full_finetune_distributed.py

```diff
@@ -623,6 +644,9 @@ def save_checkpoint(
                 epoch=epoch,
                 intermediate_checkpoint=intermediate_checkpoint,
             )
+            log.info(f"Saving checkpoint took {time.perf_counter() - start:.2f} secs")
+
+        torch.distributed.barrier()
```

Comment:
nit: Can we add a comment here for why this is necessary? Context will be lost to the ether soon.